unblob/handlers/archive/cpio.py

```diff
@@ -100,15 +100,22 @@ def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]
             else:
                 offset = round_up(file.tell(), self._PAD_ALIGN)
 
-        # Add padding that could exists between the cpio trailer and the end-of-file.
-        # cpio aligns the file to 512 bytes
-        offset = round_up(offset, self._FILE_PAD_ALIGN)
-
+        end_offset = self._pad_file(file, offset)
         return ValidChunk(
             start_offset=start_offset,
-            end_offset=offset,
+            end_offset=end_offset,
         )
 
+    @classmethod
+    def _pad_file(cls, file: File, end_offset: int) -> int:
+        """CPIO archives can have a 512 bytes block padding at the end."""
+        file.seek(end_offset, io.SEEK_SET)
+        padded_end_offset = round_up(end_offset, cls._FILE_PAD_ALIGN)
```

Comment:
I think this works for most of the inputs, as the input is most probably aligned to block boundaries.  However when the `start_offset` is not aligned to block boundary, all of the block based calculations need to take this into account, otherwise the end result would be aligned to block boundary of the input file, not to block boundary when the chunk is extracted.

Take a look at `OffsetFile` introduced in 462aa6c by @martonilles (beware, that there is a confusing amount of highly related file-like objects in `calculate_chunk` there: `file`, `offset_file`, `ar.file`).

Reply:
I followed your advice and used `OffsetFile`. I also removed unnecessary piece of code from the handler that were there due to our mis-handling of unaligned content. I also added an integration test file that starts at offset 1 to verify that we can handle alignment properly.

Reply:
Fixed the end offset calculation as discussed.